sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/ColumnStats.scala

```diff
@@ -270,8 +273,8 @@ private[columnar] final class StringColumnStats extends ColumnStats {
   }
 
   def gatherValueStats(value: UTF8String, size: Int): Unit = {
```

Comment:
we should be careful. We need to make sure the max/min states will be used properly later.

Reply:
+1 on this.
I feel like we should disable column stats initially for non-UCS_BASIC string columns. This a separate task on its own.

Reply:
I was also planning stats to be covered in some future PR. The thing is that in this code we used to explicitly call `compareTo` on UTF8String, which clearly is not allowed anymore with collations, so I had to do some tuning. I tried to follow same approach as we did for `DecimalType` in this code path.

Are there any additional tests that I can run/get stats people to review this? If I am not mistaken this should be a relatively low risk change.

Reply:
On second thought - let's reduce the blast radius. I removed the column stats changes, as you suggested.